bin/compile

```diff
@@ -29,6 +29,17 @@ function indent() {
   esac
 }
 
+# exit early because we don't need to do any work if the aptfile has no packages
```

Comment:
We're repeating these grep patterns a lot all over the place now.

I'd suggest doing this check a bit later, down before the `for DEB in "$APT_CACHE_DIR/archives/"*.deb; do` line, like this:

```
shopt -s nullglob
debs=( "$APT_CACHE_DIR/archives/"*.deb )

if ! (( ${#debs[@]} )); then
  // no packages warning and exit 0
fi

for DEB in "${debs[@]}"; do

```

Reply:
I disagree.  If there are no packages to install then we don't need to do any work including running `apt-get update` so I think the exit here makes more sense.  

And that way, when packages are installed, the `for DEB in "$APT_CACHE_DIR/archives/"*.deb` line will not cause a failure so it can remain unchanged.